### PR TITLE
Solutions

### PR DESCRIPTION
**Pull request by xanderjakeq**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,14 +30,27 @@ Edit this document to include your answers after each question. Make sure to lea
...
+.map() returns an array while .forEach does not.
+
 2. What is the difference between a function and a method?
 
+A method is a function that belongs to an object.
+
 3. What is closure?
 
+A closure is created when a function is defined within a function that uses the environment (variables or other functions) that the parent function defined. It is a combination of the child function and the environment where it was defined.
+
```
Comment (sabrinafilipelli) [inline README.md:42]: I wouldn't exactly depict is a combination of the two environments, however I see that you understand the idea.

File: README.md [hunk 2]
```diff
@@ -30,14 +30,27 @@ Edit this document to include your answers after each question. Make sure to lea
...
 
+Global binding: 'this' refers to the global object
+Implicit binding: 'this' refers to the object preceding the . when calling a function
+New binding: 'this' refers to the newly created object by a constructor
+Explicit binding: 'this' is explicitly defined when calling a function by using the .call, .apply, and .bind
+
 5. Why do we need super() in an extended class?
 
+To initialize the inherited properties from the parent class.
+
```
Comment (sabrinafilipelli) [inline README.md:53]: Hmm. Not exactly. That would be why we use extends. Let's discuss this during our one-on-one.

File: challenges/classes.js [hunk 3]
```diff
@@ -4,4 +4,42 @@
...
+    }
+    getSurfaceArea(){
+        return Math.pow(this.length, 2) * 6
+    }
+}
+
+let cube = new CubeMaker(5)
+
+console.log(cube.getVolume(), cube.getSurfaceArea())
+
```
Comment (sabrinafilipelli) [inline challenges/classes.js:44]: Good.

File: challenges/functions.js [hunk 4]
```diff
@@ -7,26 +7,54 @@
...
 
 // Explain in your own words why `nestedfunction()` can access the variable `internal`.
 
 // Explanation: 
 
+/*
+  nestedfunction is within another function creating a closure.
+  Which allows it to have access to the environment where nestedfunction 
+  was defined thus able to use the internal variable
+*/
```
Comment (sabrinafilipelli) [inline challenges/functions.js:56]: Do you mean the external variable? This is a nested function. The inner function will have access to its own variables, the outer (enclosing) function's variables, arguments and it has access to global variables. This is done by means of a scope chain. Every function so created has a scope chain associated with it which helps us determine its ability to access the value of the variables from the enclosing functions. In fact, the callback functions above are an example of closure.

File: challenges/objects-arrays.js [hunk 5]
```diff
@@ -6,29 +6,64 @@
...
+  period: 'Late Jurassic'
+})
+
 // velociraptor, carnivorous, 15kg, 1.8m, Late Cretaceious
+let vel = new Dino({
+  name: 'velociraptor',
+  diet: 'carnivorous',
+  weight: '15kg',
+  length: '1.8m',
+  period: 'Late Cretaceious'
```
Comment (sabrinafilipelli) [inline challenges/objects-arrays.js:42]: Good job! I love that you used a constructor. Clever.

File: challenges/objects-arrays.js [hunk 6]
```diff
@@ -97,23 +140,25 @@ The zoos need a list of all their animal's names (names only, not scientific) co
...
+const largerPopulation = zooAnimals.filter(animal => animal.population < 5);
 console.log(largerPopulation);
 
 /* Request 4: .reduce() 
 
 The zoos need to know their total animal population across the United States.  Find the total population from all the zoos using the .reduce() method.
 
 */
-const populationTotal = 0;
+const populationTotal = zooAnimals.reduce((total, animal) => total += animal.population,0);
```
Comment (sabrinafilipelli) [inline challenges/objects-arrays.js:161]: Nice work.

File: challenges/prototypes.js [hunk 7]
```diff
@@ -5,29 +5,45 @@
...
 /* == Step 4: Create a new object that uses CuboidMaker ==
   Create a cuboid object that uses the new keyword to use our CuboidMaker constructor
   Add properties and values of length: 4, width: 5, and height: 5 to cuboid.   
 */
 
+let cuboid = new CuboidMaker({
+  length: 4,
+  width: 5,
+  height: 5
+})
```
Comment (sabrinafilipelli) [inline challenges/prototypes.js:43]: Good job.

**Review by sabrinafilipelli — APPROVED** [review]
Good job this week, Xander. I appreciate all of the hard work you've been putting in. This week's sprint challenge displays a thorough understanding of JavaScript fundamentals.  This submission has demonstrated proficiency in creating objects, constructor functions, prototype inheritance, JavaScript classes, using a for-loop to iterate through an array, using various array methods, and a basic understanding of closures. Student effectively followed the Git Flow and got their Retrospective turned in on time. Keep up the good work!

File: README.md (hunk 2 above)
Comment (xanderjakeq) [reply README.md:53]: I thought extends provide the Parent properties but with undefined values. Super() is where the values are provided (so to uhhh hmm I'll read.)

File: challenges/functions.js (hunk 4 above)
Comment (xanderjakeq) [reply challenges/functions.js:56]: I meant the variable named "internal" should've added the quotation marks

**Pull request merged**